.github/workflows/docker-build-and-test.yml

```diff
@@ -0,0 +1,104 @@
+name: Docker Build and Test
+
+on:
+  workflow_dispatch:
+    inputs:
+      run_tests:
+        description: 'Run tests'
+        required: false
+  push:
+    branches: [ "improvement/docker-testing-workflow-branch" ]  # Isolated branch name for testing to ensure we don't affect main
+
+jobs:
+  build_and_test:
+    if: ${{ github.ref == 'refs/heads/improvement/docker-testing-workflow-branch' || github.event_name == 'workflow_dispatch' }}
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_IMAGE: aisk-unit-tests  # Updated Docker image name
+      DOCKER_TAG: ${{ github.sha }}
+      PROD_MODE: false
+      TEST_MODE: true
+      # Enable debug logging
+      ACTIONS_RUNNER_DEBUG: 'true'
+      ACTIONS_STEP_DEBUG: 'true'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Verify .dockerignore is present
+        run: |
+          if [ ! -f .dockerignore ]; then
+            echo "Error: .dockerignore file is missing."
+            exit 1
+          else
+            echo ".dockerignore file is present."
+          fi
+
+      - name: Create empty .env file
+        run: touch .env
+
+      - name: Build Docker image with BuildKit and verbose logging
+        run: |
+          echo "Building Docker image with BuildKit and verbose logging..."
+          docker build \
+            --progress=plain \
+            --build-arg PROD_MODE=$PROD_MODE \
+            --build-arg TEST_MODE=$TEST_MODE \
+            --secret id=env,src=.env \
+            -t "${DOCKER_IMAGE}:${DOCKER_TAG}" \
+            .
+
+      - name: List Docker images
+        run: docker images
+
+      - name: Create .env file with secrets
+        run: |
+          echo "Creating .env file with secrets..."
+          set +x  # Disable command echoing to protect secrets
+          {
+            echo "SAMBASTUDIO_BASE_URL=${{ secrets.SAMBASTUDIO_BASE_URL }}"
```

Comment:
Why are we using echo here, if we disable echoing?
We could simply not use the `echo` command before the assignments. We don't need it.

Reply:
It's needed because echo is used to prepare an env file to be used in the docker image, we don't want to print out sensitive keys so we disable echo and then we re-enable in order to get feedback from the rest of the script. Happy to discuss this further but I think it's needed.